compiler/rustc_hir/src/lang_items.rs

```diff
@@ -352,3 +352,153 @@ language_item_table! {
     RangeToInclusive,        sym::RangeToInclusive,    range_to_inclusive_struct,  Target::Struct;
     RangeTo,                 sym::RangeTo,             range_to_struct,            Target::Struct;
 }
+
+pub enum GenericRequirement {
+    None,
+    Minimum(usize),
+    Exact(usize),
+}
+
+impl LangItem {
+    pub fn required_generics_count(&self) -> GenericRequirement {
```

Comment:
Maybe we can make this a part of the `language_item_table!` macro to avoid listing all the language items again? I'm not sure though, so you might want to wait for someone else's optinion, I'd feel bad if you did the work to revert it :sweat_smile:

Reply:
I was thinking the same thing last night right after I submitted the PR.  I think it makes sense from a maintenance standpoint as well.